tools/check_output_test.py

```diff
@@ -11,6 +11,34 @@
 root_path = os.path.dirname(os.path.dirname(os.path.realpath(__file__)))
 tests_path = os.path.join(root_path, "tests")
 
+wildcard = "[WILDCARD]"
+
+def pattern_match(expected, actual):
+    if len(expected) == 0:
+        return actual == 0
+    if expected == wildcard:
+        return True
+    
+    parts = str.split(expected, wildcard)
+
+    if len(parts) == 1:
+        return expected == actual
+    
+    if actual.startswith(parts[0]):
+        actual = actual[len(parts[0]):]
+    else:
+        return False
+
+    for i in range(1, len(parts)):
+        if i == (len(parts)):
+            if parts[i] == "" or parts[i] == "\n":
+                return True
+        found = actual.find(parts[i])
+        if found < 0:
+            return False
+        actual = actual[(found + len(parts[i])):]
+
+    return len(actual) == 0
```

Comment:
I feel like this is sufficiently complex that it could use a unit test. (Here are some tests from the prototype: https://github.com/denoland/deno/blob/golang/util_test.go#L25-L56)

What about moving this to `tools/util.py` and we start a `tools/util_test.py` ... No need to use a testing framework or anything - `assert` will work. Then add `tools/util_test.py` to the test runner `tools/test.py` ?